### DISCUSSION TITLE
Error in code will cause the whole window to be blacked out

### DISCUSSION THREAD

**Discussion by Muhammad-Adam** [guasam/electron-react-app#48] [General]
If i clone the full version with customised titlebar, and when there is some fatal error in my code, the whole window will just black out including the titlebar. It is the same behaviour for production code.

I would like the titlebar to be at least still visible and functional if this were to happen. Is there any way?

**Comment by codesbiome** [comment]
If the `appRenderer.tsx` is failing to render the `<WindowFrame>` component, it will make the Titlebar disappear.

To keep the titlebar when something wrong with Application component, we could create dedicated roots for rendering `WindowFrame` and `Application` components. Surely it will require additional modifications in WindowFrame component to make it possible.

A rough concept idea for rendering them separately could be something like: 

```ts
// appRenderer.tsx


// Render Window Frame in DOM
const frame = <WindowFrame ... />;
createRoot(document.getElementById('window_frame')).render(frame);

// Render Application in DOM
const app = <Application ... />;
createRoot(document.getElementById('app')).render(app);
